### PR TITLE
Changed name of CRD as Hardware classification

### PR DESCRIPTION
**Pull request by Ashughorla**
- Changed name of CRD as Hardware Classification.
- Modified README.md file.
- Changed Namespace name as hardware-classification.

### CODE REVIEW

**Comment by digambar15** [comment]
Looks good

File: api/v1alpha1/hardwareclassification_types.go
Comment (maelk) [inline api/v1alpha1/hardwareclassification_types.go:40]: Shouldn't the status be declared as subresource ? And maybe you could add some short names, since that's a long name to type in kubectl

Comment (digambar15) [reply api/v1alpha1/hardwareclassification_types.go:40]: Hey maelk, we are working on introducting short names, so we will address it in next PR.

**Comment by digambar15** [comment]
I will wait for today EOD, then I will have to merge this PR as other PR's are having hard dependancy on this one.

File: api/v1alpha1/hardwareclassification_types.go
Comment (maelk) [reply api/v1alpha1/hardwareclassification_types.go:40]: ok, what about status as subresource ?

**Comment by digambar15** [comment]
./docs/api.md:3: MD013 Line length
./docs/dev-setup.md:3: MD022 Headers should be surrounded by blank lines
./docs/dev-setup.md:4: MD032 Lists should be surrounded by blank lines
./README.md:26: MD004 Unordered list style
./README.md:30: MD004 Unordered list style
./README.md:32: MD004 Unordered list style
./README.md:36: MD004 Unordered list style
./README.md:40: MD004 Unordered list style
./README.md:46: MD004 Unordered list style
./README.md:4: MD013 Line length
./README.md:5: MD013 Line length
./README.md:7: MD013 Line length
./README.md:9: MD013 Line length
./README.md:11: MD013 Line length
./README.md:26: MD013 Line length
./README.md:30: MD013 Line length
./README.md:54: MD013 Line length
./README.md:57: MD013 Line length
./README.md:1: MD022 Headers should be surrounded by blank lines
./README.md:15: MD022 Headers should be surrounded by blank lines
./README.md:16: MD032 Lists should be surrounded by blank lines

@Ashughorla  Mostly docs issues are there, please resolve.

File: api/v1alpha1/hardwareclassification_types.go
Comment (maelk) [reply api/v1alpha1/hardwareclassification_types.go:40]: @Ashughorla @digambar15  Why not declare the status as subresource ? It is assumed in PR 17 https://github.com/metal3-io/hardware-classification-controller/pull/17/files#diff-a683702516d3abc0476a39d02ce7e142R37

**Comment by maelk** [comment]
Please address the status as subresource comment before merging

File: api/v1alpha1/hardwareclassification_types.go
Comment (nehaalhat) [reply api/v1alpha1/hardwareclassification_types.go:40]: @maelk We have addressed both changes suggested by you.
- Status under resources
- Short name of hardwareClassification as hwcc and hcc

Comment (Ashughorla) [reply api/v1alpha1/hardwareclassification_types.go:40]: @maelk We have addressed both changes suggested by you in latest commit.
- Status under resources.
- Short name of hardwareClassification as hwcc and hcc

**Comment by maelk** [comment]
As discussed in #17 , where do you want to tackle the domain issue ? I would suggest to do it as part of this PR as it is part of renaming, from metal3.io.sigs.k8s.io to metal3.io . Then when this PR gets merged, you can rebase #17 on top.

**Comment by digambar15** [comment]
> As discussed in #17 , where do you want to tackle the domain issue ? I would suggest to do it as part of this PR as it is part of renaming, from metal3.io.sigs.k8s.io to metal3.io . Then when this PR gets merged, you can rebase #17 on top.

+1

**Comment by digambar15** [comment]
@maelk  Almost all the comments are resolved including changing the GroupName. Also submitted some additions.

File: api/v1alpha1/hardwareclassification_types.go
Comment (maelk) [inline api/v1alpha1/hardwareclassification_types.go]: What are the two Cs standing for ? hwc and hc might make more sense since you removed controller for the CRD name

**Review by maelk — COMMENTED** [review]
Looks good, one last nit on the short name, but otherwise that's pretty much ready to go for me

File: api/v1alpha1/hardwareclassification_types.go
Comment (digambar15) [reply api/v1alpha1/hardwareclassification_types.go]: Agree. Will change it.

**Comment by digambar15** [comment]
Thanks @Ashughorla.

**Review by maelk — APPROVED** [review]

**Pull request merged**
